src/basic/process-util.c

```diff
@@ -2054,14 +2060,57 @@ int posix_spawn_wrapper(const char *path, char *const *argv, char *const *envp,
         r = posix_spawnattr_init(&attr);
         if (r != 0)
                 return -r; /* These functions return a positive errno on failure */
-        /* Set all signals to SIG_DFL */
-        r = posix_spawnattr_setflags(&attr, POSIX_SPAWN_SETSIGMASK|POSIX_SPAWN_SETSIGDEF);
+
+#if HAVE_PIDFD_SPAWN
+        if (cgroup) {
+                r = cg_get_path_and_check(
+                                SYSTEMD_CGROUP_CONTROLLER,
+                                cgroup,
+                                /* suffix= */ NULL,
+                                &resolved_cgroup);
+                if (r < 0) {
+                        r = -r;
```

Comment:
frankly, this gets too confusing. Let's maybe rework everything to just use our usual negative error numbres if we mix regular error handling and this weird posix_spawn() handling in one function.

or with other words, change all lines like this:

```c
r = posix_spawnxyz();
if (r != 0)
        goto fail;
```

to something like this:

```c
r = -posix_spawnxyz();
if (r < 0)
        goto fail;
```

you get the idea...

and maybe this even enables you to just kill the "fail" label altogether.

Reply:
I think I prefer `return -r` as the sign change is more prominent